Backpack/FloatingNotification/Classes/BPKFloatingNotification.swift

```diff
@@ -0,0 +1,186 @@
+    private var notificationsQueue = [FloatingNotificationViewModel]()
+    
+    // MARK: Subviews
+    private let stackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.distribution = .fillProportionally
+        stackView.alignment = .top
+        stackView.axis = .horizontal
+        stackView.spacing = BPKSpacingMd
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        return stackView
+    }()
+    
+    private let iconView: BPKIconView = {
+        let iconView = BPKIconView(iconName: .none, size: .small)
+        iconView.tintColor = BPKColor.textOnDarkColor
+        iconView.isAccessibilityElement = false
+        iconView.translatesAutoresizingMaskIntoConstraints = false
+        iconView.setContentHuggingPriority(.required, for: .horizontal)
+        iconView.setContentCompressionResistancePriority(.required, for: .horizontal)
+        return iconView
+    }()
+    
+    private let messageLabel: BPKLabel = {
+        let label = BPKLabel()
+        label.fontStyle = .textFootnote
+        label.textColor = BPKColor.textOnDarkColor
+        label.textAlignment = .natural
+        label.numberOfLines = 0
+        label.lineBreakMode = .byWordWrapping
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.setContentHuggingPriority(.defaultLow, for: .horizontal)
+        return label
+    }()
+    
+    private let button: BPKButton = {
+        let button = BPKButton(size: .default, style: .primary)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setContentHuggingPriority(.required, for: .horizontal)
+        button.setContentCompressionResistancePriority(.required, for: .horizontal)
+        return button
+    }()
+    
+    // MARK: Initialisation
+    public init() {
+        super.init(frame: .zero)
+        button.addTarget(self, action: #selector(buttonTapped), for: .touchUpInside)
+        
+        animator.delegate = self
+        alpha = 0
+        backgroundColor = BPKColor.corePrimaryColor
+        layer.cornerRadius = BPKCornerRadiusMd
+        translatesAutoresizingMaskIntoConstraints = false
+        addSubviews()
+        BPKShadow.shadowLg().apply(to: layer)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func addSubviews() {
+        [iconView, messageLabel, button].forEach { stackView.addArrangedSubview($0) }
+        addSubview(stackView)
+    }
+    
+    // MARK: Animation & rendering
+    public func show(_ viewModel: FloatingNotificationViewModel) {
+        notificationsQueue.append(viewModel)
+        renderNext()
+    }
+    
+    private func renderNext() {
+        if let nextViewModel = notificationsQueue.first, !animator.isNotificationDisplayed {
+            render(viewModel: nextViewModel)
+            animator.animateUp(hideAfter: nextViewModel.hideAfter)
```

Comment:
When VoiceOver is on, the floating notification should not hide as long as it has focus. 
(Check out what the Youtube app does when you save/unsave a video)

Reply:
This one was tricky but added it in